docs/adr/0001-daily-active-use-server-side-metrics-glean.md

```diff
@@ -0,0 +1,121 @@
+* A collection of metrics, emitted as a single "Ping Event" make querying of related data simpler.
+* Core of Glean's purpose is to measure user interactions and the rich metadata that accompanies it.
+* Capacity for future expansion of application metrics within Sync, beyond DAU.
+* Prepares for implementation of same measurements in autopush, also using Glean.
+* Easier to query, have data team support to set up queries.
+* Use of standardized Mozilla tooling.
+* Establishment of team knowledge of using Glean.
+* Establishment of server-side Rust best practices, leading to easier development for backend Rust applications.
+* Transparency in data review process with consideration made to minimum collection of data.
+
+#### Cons
+
+* The `glean_parser` tool used by other Glean applications is currently not supported for Rust. Furthermore, client applications can use the Glean SDK and this is also not an option for us.
+* Server side metrics have not yet been implemented for a Rust server application of this kind, so this is new territory.
+* There is added complexity of data review process and registration of the application to Glean's probe scraper, PubSub topics, ingestion pipeline etc.
+* May require more uptfront development effort to get up to speed with new code and process.
+* Some concerns exist around volume of data emitted from the service and if it is feasible, but we won't know until we try.
+
+### B. Custom Glean Implementation 
+
+This was originally appearing to be the desired approach to measuring DAU via Glean.  This was predicated on the ease of prototyping a custom implementation that imitated the Glean team's logic to create "Glean-compliant" output that could be configured to be ingested.  However, in consulting with the Glean team and evaluating the pros and cons of this approach, it became clear this approach had considerably more drawbacks than implementing the `glean_parser` for Rust. These drawbacks were a lack of testing, validation, less support from the Glean team, and the potential problems with maintenance and adding Glean metrics in the future.
+
+#### Pros
+* Gives team control over implementation and allows us to customize the Glean logging as we see fit.
+* Does not require contribution to Glean team's repos, which potentially limits the scope understanding a new codebase.
+* Easy to prototype and make changes.
+* Doesn't require understanding the templating logic and libraries in the `glean_parser`
+
+#### Cons
+* There is no built-in testing suite or validation, so this would put a larger development burden on us and require the Glean team's review.
+* Lack of testing and validation means higher likelihood of bugs.
+* If we decide to add new Glean metrics in the future, this may break the custom implementation and impose a greater maintenance overhead.
+* Time required to understand the Glean team's implementations anyways, in order to replicate behavior and data structures.
```

Comment:
nit: "anyways" bothers me. But I think the larger point here is that we still have the overhead of needing to understand the Glean team's implementations.